sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryPartCTests.cs

```diff
@@ -0,0 +1,244 @@
+        {
+            using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+            using var activity = activitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Client,
+                parentContext: default,
+                startTime: DateTime.UtcNow);
+
+            var telemetryPartBRemoteDependencyData = TelemetryPartB.GetRemoteDependencyData(activity);
+            Assert.False(telemetryPartBRemoteDependencyData.Properties.TryGetValue(msLinks, out var mslinks));
+        }
+
+        [Fact]
+        public void RemoteDependencyDataPropertiesContainsMSLinksWhenActivityHasLinks()
+        {
+            using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+            ActivityLink activityLink = new ActivityLink(new ActivityContext(
+                ActivityTraceId.CreateRandom(),
+                ActivitySpanId.CreateRandom(),
+                ActivityTraceFlags.None), null);
+
+            List<ActivityLink> links = new List<ActivityLink>();
+            links.Add(activityLink);
+
+            using var activity = activitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Client,
+                parentContext: default,
+                null,
+                links,
+                startTime: DateTime.UtcNow);
+
+            string expectedMSlinks = GetExpectedMSlinks(links);
+            var telemetryPartBRemoteDependencyData = TelemetryPartB.GetRemoteDependencyData(activity);
+
+            Assert.True(telemetryPartBRemoteDependencyData.Properties.TryGetValue(msLinks, out var actualMSlinks));
+            Assert.Equal(expectedMSlinks, actualMSlinks);
+        }
+
+        [Fact]
+        public void LinksAreTruncatedWhenCannotFitInMaxLength()
+        {
+            using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+            List<ActivityLink> links = new List<ActivityLink>();
+
+            int numberOfLinks = 150; //arbitrary number > 100
+            int maxLinksAllowed = 100;
+            int maxLength = 8192;
+
+            for (int i = 0; i < numberOfLinks; i++)
+            {
+                ActivityLink activityLink = new ActivityLink(new ActivityContext(
+                ActivityTraceId.CreateRandom(),
+                ActivitySpanId.CreateRandom(),
+                ActivityTraceFlags.None), null);
+                links.Add(activityLink);
+            }
+
+            string expectedMSlinks = GetExpectedMSlinks(links.GetRange(0, maxLinksAllowed));
+
+            using var activity = activitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Client,
+                parentContext: default,
+                null,
+                links,
+                startTime: DateTime.UtcNow);
+
+            // Only checking dependency here. The flow will be same for other telemetry types.
```

Comment:
you can parameterize the test and check both types, without duplicating a lot of code.